projects/gloo/pkg/plugins/kubernetes/uds_convert_test.go

```
@@ -154,6 +159,176 @@ var _ = Describe("UdsConvert", func() {
 				Expect(upstreamConfigJson).To(Equal(expectedCfgJson))
 			}
 
+			CreateUpstreamWithSpec := func(uc *KubeUpstreamConverter, ctx context.Context, svc *kubev1.Service, port kubev1.ServicePort, upstream *v1.Upstream) *v1.Upstream {
+				for _, sc := range uc.serviceConverters {
+					if err := sc.ConvertService(svc, port, upstream); err != nil {
+						contextutils.LoggerFrom(ctx).Errorf("error: failed to process service options with err %v", err)
```

can we just return an error here?

---

Resolved here: https://github.com/solo-io/gloo/pull/6036/commits/fd8078fe46999a8b0344b63828a0c0f2cdea47cc